packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -799,7 +1007,7 @@ function parseSlot(
 ): Slot {
     const location = ast.sourceLocation(parse5ElmLocation);
 
-    const hasDirectives = ctx.findSibling(ast.isForBlock) || ctx.findSibling(ast.isIf);
+    const hasDirectives = ctx.findInCurrentElementScope(ast.isElementDirective);
```

Comment:
Include the new directives when checking slots.

Note: this restriction may be relaxed in a future PR for lwc:if, lwc:elseif, and lwc:else directives.